workflow/executor.go

```diff
@@ -313,10 +313,9 @@ func (e *executor) processSteps(
 	return runnableSteps, stepOutputProperties, stepLifecycles, stepRunData, nil
 }
 
-func applyLifecycleScopes(
-	stepLifecycles map[string]step.Lifecycle[step.LifecycleStageWithSchema],
-	typedInput schema.Scope,
-) error {
+// BuildNamespacedScopes creates a map of a workflow's steps' lifecycles to
+// the object schemas that exist during that lifecycle.
```

Comment:
I think there should be more context about it. You say "map of a workflow's steps' lifecycles to ...". I think it would make more sense to talk about how it maps the step's path and ID to a map of object IDs to object schemas.